File: 29-08-2024_find_length_of_loop.py
class Solution:
    def find_loop_length(self, start):
        current_node = start
        visited_nodes = set()

        while current_node is not None:
            if current_node in visited_nodes:
                loop_length = 1
                temp_node = current_node.next
                while temp_node != current_node:
                    loop_length += 1
                    temp_node = temp_node.next
                return loop_length

            visited_nodes.add(current_node)
            current_node = current_node.next

        return 0

File: 29-08-2024_find_length_of_loop.cpp
#include <unordered_set>

class Solution {
public:
    int findLoopLength(Node* start) {
        Node* currentNode = start;
        std::unordered_set<Node*> visitedNodes;

        while (currentNode != nullptr) {
            if (visitedNodes.find(currentNode) != visitedNodes.end()) {
                int loopLength = 1;
                Node* tempNode = currentNode->next;
                while (tempNode != currentNode) {
                    loopLength++;
                    tempNode = tempNode->next;
                }
                return loopLength;
            }
            visitedNodes.insert(currentNode);
            currentNode = currentNode->next;
        }
        return 0;
    }
};
